app/modules/gpio.c

```diff
@@ -170,42 +171,95 @@ static int lgpio_write( lua_State* L )
 // gpio.mode(1,gpio.OUTPUT,gpio.PULLUP)
 // gpio.serout(1,0,{20,10,10,20,10,10,10,100}) -- sim uart one byte 0x5A at about 100kbps
 // gpio.serout(1,1,{8,18},8) -- serial 30% pwm 38k, lasts 8 cycles
+
+typedef struct
+{
+  unsigned pin;
+  unsigned level;
+  uint32 index;
+  uint32 repeats;
+  uint32 *delay_table;
+  uint32 tablelen;
+  lua_State* L;
+  int lua_done_ref; // callback when transmission is done
+} serout_t;
+static serout_t serout;
+static const os_param_t TIMER_OWNER = 0x6770696f; // "gpio"
+
+static void seroutasync_done (task_param_t arg)
+{
+  lua_rawgeti (serout.L, LUA_REGISTRYINDEX, serout.lua_done_ref);
+  lua_call (serout.L, 0, 0);
```

Comment:
```lua
if (lua_pcall(serout.L, 0, 0, 0)) {
  // Uncaught Error. Print instead of sudden reset
  error(serout.L, "error: %s", lua_tostring(serout.L, -1));
}
```